x-pack/plugins/actions/server/builtin_action_types/servicenow/types.ts

```diff
@@ -4,18 +4,97 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export {
-  ExternalIncidentServiceConfiguration as ServiceNowPublicConfigurationType,
-  ExternalIncidentServiceSecretConfiguration as ServiceNowSecretConfigurationType,
-} from '../case/types';
+/* eslint-disable @typescript-eslint/no-explicit-any */
 
-export interface CreateIncidentRequest {
-  summary: string;
-  description: string;
-}
+import { TypeOf } from '@kbn/config-schema';
+import {
+  ExternalIncidentServiceConfigurationSchema,
+  ExternalIncidentServiceSecretConfigurationSchema,
+  ExecutorParamsSchema,
+  ExecutorSubActionPushParamsSchema,
+  ExecutorSubActionGetIncidentParamsSchema,
+  ExecutorSubActionHandshakeParamsSchema,
+} from './schema';
+import { ActionsConfigurationUtilities } from '../../actions_config';
+import { IncidentConfigurationSchema } from './case_shema';
+import { PushToServiceResponse } from './case_types';
 
-export type UpdateIncidentRequest = Partial<CreateIncidentRequest>;
+export type ServiceNowPublicConfigurationType = TypeOf<
+  typeof ExternalIncidentServiceConfigurationSchema
+>;
+export type ServiceNowSecretConfigurationType = TypeOf<
+  typeof ExternalIncidentServiceSecretConfigurationSchema
+>;
 
 export interface CreateCommentRequest {
   [key: string]: string;
 }
+
+export type ExecutorParams = TypeOf<typeof ExecutorParamsSchema>;
+export type ExecutorSubActionPushParams = TypeOf<typeof ExecutorSubActionPushParamsSchema>;
+
+export type IncidentConfiguration = TypeOf<typeof IncidentConfigurationSchema>;
+
+export interface ExternalServiceCredentials {
+  config: Record<string, unknown>;
+  secrets: Record<string, unknown>;
+}
+
+export interface ExternalServiceValidation {
+  config: (configurationUtilities: ActionsConfigurationUtilities, configObject: any) => void;
+  secrets: (configurationUtilities: ActionsConfigurationUtilities, secrets: any) => void;
+}
+
+export interface ExternalServiceIncidentResponse {
+  id: string;
+  title: string;
+  url: string;
+  pushedDate: string;
+}
+
+export interface ExternalServiceParams {
```

Comment:
Wonder if we should even have this type, and instead use `Record<string, unknown>` where it's used.  It looks a bit funny when used as the result type of `getIncident()` and `findIncidents()` below